src/map/unit.c

```diff
@@ -488,102 +489,142 @@ static int unit_walktoxy_timer(int tid, int64 tick, int id, intptr_t data)
 	return 0;
 }
 
+/**
+ * Timer for delayed execution of unit->walktoxy once triggered
+ * @param tid: Timer ID, unused
+ * @param tick: Tick, unused
+ * @param id: ID of block_list to execute the action
+ * @param data: uint32 data cast to intptr_t with x-coord in lowest 16 bits and y-coord in highest 16 bits
+ * @return 0: success, 1: failure
+ */
 static int unit_delay_walktoxy_timer(int tid, int64 tick, int id, intptr_t data)
 {
 	struct block_list *bl = map->id2bl(id);
-
-	if (!bl || bl->prev == NULL)
-		return 0;
-	unit->walktoxy(bl, (short)((data>>16)&0xffff), (short)(data&0xffff), 0);
-	return 1;
+	nullpo_retr(1, bl);
+	nullpo_retr(1, bl->prev);
+	short x = (short)GetWord((uint32)data, 0);
+	short y = (short)GetWord((uint32)data, 1);
+	unit->walktoxy(bl, x, y, 0);
+	return 0;
 }
 
-//flag parameter:
-//&1 -> 1/0 = easy/hard
-//&2 -> force walking
-//&4 -> Delay walking if the reason you can't walk is the canwalk delay
-//&8 -> Search for an unoccupied cell and cancel if none available
+/**
+ * Makes a unit walk to (x, y) coordinates
+ * @param bl: block_list of unit to move
+ * @param x: x-coordinate
+ * @param y: y-coordinate
+ * @param flag: flag paramater with following options:
+ *  - `& 1` -> 1/0 = easy / hard
+ *  - `& 2` -> Force walking
+ *  - `& 4` -> Delay walking, if the reason you can't walk is the `canwalk delay`
+ *  - `& 8` -> Search for an unoccupied cell and cancel if none available
+ * .
+ * @return 0: success, 1: failure
+ */
 static int unit_walktoxy(struct block_list *bl, short x, short y, int flag)
 {
+	// TODO: change flag to enum? [skyleo]
 	struct unit_data* ud = NULL;
 	struct status_change* sc = NULL;
 	struct walkpath_data wpd;
 
-	nullpo_ret(bl);
+	nullpo_retr(1, bl);
 
 	ud = unit->bl2ud(bl);
 
-	if( ud == NULL) return 0;
+	nullpo_retr(1, ud);
 
-	if (battle_config.check_occupied_cells && (flag&8) && !map->closest_freecell(bl->m, bl, &x, &y, BL_CHAR|BL_NPC, 1)) //This might change x and y
-		return 0;
+	if ((flag & 8) != 0 && battle_config.check_occupied_cells != 0) {
+		if (!map->closest_freecell(bl->m, bl, &x, &y, BL_CHAR | BL_NPC, 1)) // This might change x and y
```

Comment:
perhaps you can just add new line with && instead of second if?

Reply:
I did this due to the code style rules of multiple-line if-conditions, that the previous line has to be bigger than the next one. Which is possible to do here, but I think code-readability improves a lot from having the flag and configurational checks first, if possible.

Usually when you look for a certain flag-setting you skim through the very left of every line, expecting it to be there.

Reply:
Unresolve if my excuse isn't convincing enough.